changelogs/unreleased/popen-wait-busy-loop.md

```diff
@@ -0,0 +1,5 @@
+## feature/lua/popen
+
+* Eliminate polling in `<popen handle>:wait()` (gh-4915).
+
```

Comment:
Nit: AFAIU a changelog entry shouldn't contain empty lines.

Reply:
In my understanding a changelog entry is just list item formatted in Markdown (GFM, to be precise). I use the Markdown syntax for multiparagraph list items:

```markdown
* Concise description (gh-xxxx).

  Details that are valuable in context of a changelog: what becomes better for
  a user, which use cases are affected and so on.
```

I don't see why we should forbid it.

Ideally every change can be described with just one line, but sometimes I want to add a bit more. Say, here I describe the same, but in simpler words.

Reply:
I wouldn't put any details in a changelog - there's a link to the ticket for that. With an empty line, a changelog looks kinda torn. I don't insist though.

Reply:
A changelog with too much details looks bloated and difficult to read. The goal of a changelog is to highlight changes visible to users so that one can quickly look through it and understand how it affects their setup. Details may be interesting, but they should be described somewhere else (tickets, PRs, commit log). I think of a changelog as of a ToC so one-two sentences per entry should be enough. This is IMO; feel free to ignore.

Reply:
> and understand how it affects their setup

It is why I have added the second sentence ('Now it reacts to SIGCHLD faster and performs less unnecessary work'). It seems, we are on the same position regarding what should be in the changelog, but argue regarding the best formatting.

> one-two sentences per entry should be enough

It is what I have there: two sentences. I can reformat it this way:

```markdown
## feature/lua/popen

* Eliminated polling in `<popen handle>:wait()`, so now it reacts to SIGCHLD
  faster and performs less unnecessary work (gh-4915).
```

I don't like two sentences in a line, because I don't know how to place `(gh-xxxx)` this way:

```markdown
* Blah blah. Blah blah (gh-xxxx).
```

```markdown
* Blah blah (gh-xxxx). Blah blah.
```

```markdown
* Blah blah. Blah blah. (gh-xxxx)
```

Feels irregular.

Reply:
I typically place (gh-XXXX) after the last stentence:

```
* Fixed a bug that broken everything. Now, everything works just fine (gh-XXXX).
```

Looks like this is in accordance with our [guideline](https://www.tarantool.io/en/doc/latest/contributing/release_notes/).

> Feels irregular.

We desperately need an [auto-checker](https://github.com/tarantool/tarantool/issues/6996) for changelogs :-/

Reply:
I mean irregular in a grammatical sense. `(gh-xxxx)` is about both sentences, but is added as part of the second. That's feels strange.

Anyway, I reworded it as one sentence.